src/connector/builder.rs

```diff
@@ -246,9 +267,19 @@ impl ConnectorBuilder<WantsProtocols1> {
     /// If this method is called, hyper-rustls will instead verify that server
     /// certificate contains `override_server_name`. Domain name included in
     /// the URL will not affect certificate validation.
-    pub fn with_server_name(mut self, override_server_name: String) -> Self {
-        self.0.override_server_name = Some(override_server_name);
-        self
+    #[deprecated(since = "0.27.1", note = "use Self::with_server_name_resolver instead")]
```

Comment:
I think it'd be helpful to mention the fixed name resolver here since that's the way to achieve the replacement of `with_server_name` with minimal changes:

```suggestion
    #[deprecated(since = "0.27.1", note = "use Self::with_server_name_resolver with FixedServerNameResolver instead")]
```

Reply:
> rustls / Format (pull_request) Failing after 17s

Sorry, looks like my suggested fix needs to go through `rustfmt` to appease CI.

Reply:
fixed!